.github/workflows/pr-test.yml

```diff
@@ -0,0 +1,45 @@
+name: PR Testing
+
+on: [pull_request]
+
+jobs:
+  make-workflows:
+    runs-on: ubuntu-latest
+
+    env:
+      DESTDIR: /tmp/nilrt-snac
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Fetch all history so that git-describe works
```

Comment:
Do we need to override `fetch-depth`? What's the problem if we don't override it?

Reply:
When the PR test runs `make dist`, the [`git describe` operation](https://github.com/ni/nilrt-snac/blob/master/Makefile#L9) in the Makefile tries to determine the correct project version based on the git history - generally looking backwards in the history for the latest annotated commit tag and then applying the number of commits since that tag upon it. That operation doesn't work when using the default `actions/checkout` setting: to only checkout the latest git object, without history.

If we don't set this option, all `make` operations in the PR test will print a warning from git that it cannot "describe" the repo. And the distribution tar-ball created by `make dist` will be unversioned, like `nilrt-snac-.tar.gz`.